### DISCUSSION TITLE
How do I remap navigating through the suggestions?

### DISCUSSION THREAD

**Discussion by martin-braun** [NvChad/NvChad#1598] [Q&A] [answered]
Hi,

I'd like to navigate through the suggestions using the arrow keys (up, down) instead of Tab / S-Tab.

I'm talking about these actions: 

![2022-09-26 2 05 58 PM](https://user-images.githubusercontent.com/24259245/192272366-69ffd7b7-c701-46bc-b0ca-12060cc2c336.gif)
(I know the `:` is wrong here btw)

Could anybody please tell me how I can figure this out? I suppose, I need to update the mappings of the particular plugin in the chadrc.lua, but I have hard times figuring this out. 

The default mappings.lua only shows the Tab bindings for the buffer navigation.

**Comment by siduck** [comment, marked as answer]
Tab & shift + tab or Ctrl n & Ctrl p

**Reply by martin-braun** [reply]
Sorry, my goal is to use arrow keys instead. Could you please tell me how I can map those properly?

**Reply by siduck** [reply]
@martin-braun override cmp config , read through it and replace C-n etc with <Up> <Down> 

Also using arrow keys will slow you down. It goes against vim workflow too

**Reply by martin-braun** [reply]
@siduck First of all, thank you! 

I know that it is slower to use arrow keys. The reason why I want to map arrow keys is because I use karabiner to map fn+h/j/k/l to arrow keys system-wide on my MacBook. So actually, I use vim controls everywhere and I enjoy to give the fn key a function in `nvim` as well. Since native drop-downs also react to fn+j / fn+k on my system it feels more consistent to go this route.

For reference, here is my solution: 

```lua
M.plugins = {
    ...
    ["hrsh7th/nvim-cmp"] = {
        override_options = function()
            local cmp = require("cmp")
            return {
                mapping = {
                    ["<C-p>"] = {},
                    ["<Up>"] = cmp.mapping.select_prev_item(),
                    ["<C-n>"] = {},
                    ["<Down>"] = cmp.mapping.select_next_item(),
                }
            }
        end
    },
    ...
}
```
